crates/miden-tx/src/errors/mod.rs

```diff
@@ -69,6 +73,28 @@ pub enum TransactionExecutorError {
     Unauthorized(Box<TransactionSummary>),
 }
 
+// NOTE CONSUMPTION ERROR
+// ================================================================================================
+
+#[derive(Debug, Error)]
+pub enum NoteConsumptionError {
+    #[error("note incompatible with target account")]
+    AccountCompatibilityError(Note),
+    #[error("error occurred while transactions were executed")]
+    ExecutionError {
```

Comment:
nit: I would change the message to something like "error occurred while consuming the note in a transaction".